src/Database/Type/DateTimeType.php

```diff
@@ -408,17 +420,46 @@ public function useMutable()
 
     /**
      * Converts a string into a DateTime object after parsing it using the locale
-     * aware parser with the specified format.
+     * aware parser with the format set by `setLocaleFormat()`.
      *
      * @param string $value The value to parse and convert to an object.
-     * @return \Cake\I18n\Time|null
+     * @return \Cake\I18n\I18nDateTimeInterface|null
+     */
+    protected function _parseLocaleValue(string $value)
+    {
+        /** @var \Cake\I18n\I18nDateTimeInterface $class */
+        $class = $this->_className;
+
+        return $class::parseDateTime($value, $this->_localeMarshalFormat);
+    }
+
+    /**
+     * Converts a string into a DateTime object after parsing it using the
+     * formats in `_marshalFormat`.
+     *
+     * @param string $value The value to parse and convert to an object.
+     * @return \DateTimeInterface|null
      */
     protected function _parseValue(string $value)
     {
-        /** @var \Cake\I18n\Time $class */
+        /** @var \DateTime|\DateTimeImmutable $class */
         $class = $this->_className;
 
-        return $class::parseDateTime($value, $this->_localeFormat);
+        foreach ((array)$this->_marshalFormat as $format) {
+            try {
+                $dateTime = $class::createFromFormat($format, $value);
+                // Check for false in case DateTime is used directly
+                if ($dateTime !== false) {
+                    return $dateTime;
+                }
+            } catch (InvalidArgumentException $e) {
```

Comment:
Unfortunately, Chronos throws this exception instead of returning false as the base class does.

https://github.com/cakephp/chronos/blob/af24b6046827491fb4d10048e42c1fb4a671d840/src/Traits/FactoryTrait.php#L214